ambry-cloud/src/main/java/com/github/ambry/cloud/VcrReplicationManager.java

```diff
@@ -168,6 +159,10 @@ public VcrReplicationManager(VerifiableProperties properties, StoreManager store
     this.cloudDestination.getTableClient(this.azureCloudConfig.azureTableNameCorruptBlobs);
     azureTableNameReplicaTokens = this.azureCloudConfig.azureTableNameReplicaTokens;
     this.cloudDestination.getTableClient(azureTableNameReplicaTokens);
+    String datacenter = clusterMap.getDatacenterName(clusterMap.getLocalDatacenterId());
+    logger.info("Local datacenter id = {}, Local datacenter name = {}", clusterMap.getLocalDatacenterId(), datacenter);
+    replicaThreadPoolByDc.put(datacenter,
+        createThreadPool(datacenter, getNumReplThreads(cloudConfig.backupNodeCpuScale), true));
```

Comment:
I understand what you are trying to do here. VCR should only replicate from local datacenter. And you only create threads for local datacenter.

But we are using a configuration to control which datacenter to replicate from for vcr, which should only contains local datacenter in configuration. I feel that we should just remove the configuration and only rely on the logic where we just need to replicate from local datacenter. 

At least let's just put a comment here saying that there can't be any other datacenter in the configuration, otherwise, vcr won't run.

Reply:
got it !

Reply:
done